src/Microsoft.Deployment.DotNet.Releases/src/Product.cs

```diff
@@ -0,0 +1,212 @@
+        public ReleaseVersion LatestRuntimeVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The version of the SDK included in the latest release.
+        /// </summary>
+        /// <remarks>
+        /// This is usually the SDK with the highest feature band. A <see cref="ProductRelease"/>
+        /// may include multiple SDKs across different feature bands, all of which carry the same runtime version.
+        /// </remarks>
+        [JsonProperty(PropertyName = "latest-sdk")]
+        public ReleaseVersion LatestSdkVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The name of the product.
+        /// </summary>
+        [JsonProperty(PropertyName = "product")]
+        public string ProductName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The URL pointing to the releases.json file that contains information about all the releases 
+        /// associated with this <see cref="Product"/>.
+        /// </summary>
+        [JsonProperty(PropertyName = "releases.json")]
+        public Uri ReleasesJson
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The support phase of the Product. For an LTS release, the <see cref="EndOfLifeDate"/> property should 
+        /// be checked to confirm whether a release is still supported. 
+        /// </summary>
+        /// <remarks>
+        /// The EOL dates are often published in advance, but there can be delays to updating the support phase in the published
+        /// data.
+        /// </remarks>
+        [JsonProperty(PropertyName = "support-phase")]
+        public SupportPhase SupportPhase
+        {
+            get;
+            private set;
+        } = SupportPhase.Unknown;
+
+        /// <summary>
+        /// Gets a collection of all releases associated with this <see cref="Product"/>.
+        /// </summary>
+        /// <returns>A collection of all releases for this product.</returns>
+        public async Task<ReadOnlyCollection<ProductRelease>> GetReleasesAsync()
+        {
+            return await GetReleasesAsync(ReleasesJson, this);
+        }
+
+        /// <summary>
+        /// Gets a collection of all releases associated with this <see cref="Product"/> using a file
+        /// containing the releases data.
+        /// </summary>
+        /// <param name="path">The path of the file containing the releases data.</param>
+        /// <param name="downloadLatest">When <see langword="true"/>, the latest copy of the releases data is used
+        /// if the online version is newer than the local file copy.</param>
+        /// <returns>A collection of releases associated with this <see cref="Product"/>.</returns>
+        public async Task<ReadOnlyCollection<ProductRelease>> GetReleasesAsync(string path, bool downloadLatest)
```

Comment:
One scenario would be where you have a tool that can run offline (so you have a static set of files), but when running online, it looks at the files on disk (if they exist), updates them if there are newer JSON files or updates them if they don't exist and you're going online

Reply:
In this usage scenario, the caller needs to check the online status because they need to specify whether or not `downloadLatest` can be performed.  Are we providing justifiable value with this method versus one that simply takes a file path?  What is the determining factor on where the convenience line is drawn? Would it be better to hold off adding this overload based on user feedback?

Reply:
We already have internal tooling leveraging this functionality (more than one application and a number of prototype tools). All we know is we have customers consuming the data, but we know very little of the specific issues they've tried to solve. I don't think we have any guidelines laid out.

We also saw issues when the service can be overwhelmed with requests, so being able to work offline and only occasionally go online is useful.